stores/sql/main.go

```diff
@@ -244,6 +244,45 @@ func ContractRoots(ctx context.Context, tx sql.Tx, fcid types.FileContractID) ([
 	return roots, nil
 }
 
+// ContractRootsDiff returns the indices, offset by the given offset, of roots
+// that are not in the contract.
+func ContractRootsDiff(ctx context.Context, tx sql.Tx, fcid types.FileContractID, roots []types.Hash256, offset uint64) (indices []uint64, err error) {
+	// build select query
+	query := "SELECT ? as idx, ? as root" + strings.Repeat(" UNION ALL SELECT ? AS idx, ?", len(roots)-1)
```

Comment:
Have you tried this on a larger DB? My concern is that this might be slower due to the lack of usable indices when joining than just selecting all roots where `c.fcid = ? AND s.root IN (roots...)` and then doing something like this

```go
rootIdx := make(map[types.Hash256]int) // populated from 'roots' input
slices.Sort(fetchedRoots, func(i, j int) bool {
  return rootIdx[fetchedRoots[i]] < rootIdx[fetchedRoots[j]]
})
```

Sure in that case you fetch roots instead of indices, but since we only fetch missing roots, being able to use an index might still make a big difference on a large db.

Reply:
Whatever is faster on a node like Arequipa is fine by me. Just need to make sure we also pick a large contract to also reach that 25600 root batch size.

Reply:
See PR description.

Reply:
200MB takes 11 seconds? That feels like it's way too long for diffing about 8MB of roots and removing them from the contract.